ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/translator/RexNodeConverter.java

```diff
@@ -936,14 +678,10 @@ protected RexNode convert(ExprNodeConstantDesc literal) throws CalciteSemanticEx
       break;
     case TIMESTAMPLOCALTZ:
       final TimestampString tsLocalTZString;
-      if (value == null) {
-        tsLocalTZString = null;
-      } else {
-        Instant i = ((TimestampTZ)value).getZonedDateTime().toInstant();
-        tsLocalTZString = TimestampString
-            .fromMillisSinceEpoch(i.toEpochMilli())
-            .withNanos(i.getNano());
-      }
+      Instant i = ((TimestampTZ)value).getZonedDateTime().toInstant();
```

Comment:
I'm not sure; but the null check seemed to be needed here.
or...there can't be a "null" which is typed as TIMESTAMPLOCALTZ?

Reply:
`value` cannot be null if we are here (note the check in L569). For the rest of types, sometimes there is a check on null value indeed, but it is because value is not final and it is replaced within those case blocks.